jerry-core/ecma/builtin-objects/ecma-builtin-json.c

```diff
@@ -1118,32 +1068,8 @@ ecma_builtin_json_stringify (ecma_value_t this_arg, /**< 'this' argument */
 
     if (ecma_is_value_empty (ret_value))
     {
-      /* 9. */
-      ecma_object_t *obj_wrapper_p = ecma_op_create_object_object_noarg ();
-      ecma_string_t *empty_str_p = ecma_get_magic_string (LIT_MAGIC_STRING__EMPTY);
-
-      /* 10. */
-      ecma_value_t put_comp_val = ecma_op_object_put (obj_wrapper_p,
-                                                      empty_str_p,
-                                                      arg1,
-                                                      false);
-
-      JERRY_ASSERT (ecma_is_value_true (put_comp_val));
-      ecma_free_value (put_comp_val);
-
-      /* 11. */
-      ECMA_TRY_CATCH (str_val,
-                      ecma_builtin_json_str (empty_str_p, obj_wrapper_p, &context),
-                      ret_value);
-
-      ret_value = ecma_copy_value (str_val);
-
-      ECMA_FINALIZE (str_val);
-
-      ecma_deref_object (obj_wrapper_p);
+      ret_value = ecma_builtin_json_string_from_object (arg1);
```

Comment:
I think you organized out this function badly. For example you ignore that maybe the context was previously set. Furthermore the tests/jerry/json-stringify.js also fails.